scipy/special/_logsumexp.py

```diff
@@ -97,7 +98,10 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
             a = a + 0.  # promote to at least float
             a[b == 0] = -np.inf
 
-    a_max = np.amax(a, axis=axis, keepdims=True)
+    if np.issubdtype(a.dtype, np.complexfloating):
+      a_max = np.amax(abs(a), axis=axis, keepdims=True)
```

Comment:
Had to think this through: the goal of `a_max` is to stop overflow, for which only `a.real.max()` matters. In principle, large imaginary numbers are of course problematic too, since then the argument becomes indeterminate. But not obvious it helps to subtract that from `a` before doing the exponential. So, I *think* this can be `a_max = np.amax(a.real, ...)` unconditionally (with a comment stating we do `.real` for complex perhaps).

Separately: the indentation is off here.

Reply:
The reason I changed this is because if `a_max` is complex, it affects the phase and so the sign computation below does not return a useful result.

Reply:
I think since the magnitude of the exponential is dictated by the real part, and the phase is dictated by the imaginary part, you're right that `a.real` probably makes more sense.